src/libaktualizr/uptane/fetcher.cc

```diff
@@ -49,32 +49,54 @@ static size_t DownloadHandler(char* contents, size_t size, size_t nmemb, void* u
     auto event = std::make_shared<event::DownloadProgressReport>(ds->target, "Downloading", calculated);
     (*(ds->events_channel))(event);
   }
+  if (ds->fetcher->isPaused()) {
+    return written_size + 1;
```

Comment:
Sorry, could you add clarifying comments here?

Reply:
We return incorrect size because we need stop download. libcurl has some magic value we could return to pause download (https://curl.haxx.se/libcurl/c/curl_easy_pause.html) and I tried this approach but I rejected it because it keeps connection alive and after some time server drops this connection with timeout error.

Reply:
Yes, looks like the best way to abort the installation. It would be nice to have the comment in the code though.